### ISSUE TITLE
Binance-api has a limited amount of candlesticks

### ISSUE THREAD

**Issue by PanderBoy18** [jaggedsoft/php-binance-api#200]
**Binance-api has a limited amount of candlesticks**

Dear Jaggedsoft,

I'm struggling with a problem! I am trying to get all the candlesticks from september of the LTC 15M, but It looks like there is some kind of limit of candlesticks (500). This is the call I am using:
`$binance_api->chart(["LTCUSDT"], "15m", function($binance_api, $symbol, $chart)`

I need the whole month of september, but it doesn't get further than: 2018-10-22 08:15:00 (Its now 2018-10-27 13:05:00), So thats a exactly amount of 500 Candlesticks. Can I bypass this limit or something?

**Comment by jaggedsoft** [comment]
Unfortunately this is rather limited. Using the candlesticks function we can increase our limit to 1000 and specify a start and end time, so worst case scenario we could make several requests.
If you need longer term historical data, it might be easier to grab it from cryptocompare, this is the same API that powers https://info.binance.com

```php
     * @param $symbol string to query
     * @param $interval string to request
     * @param $limit int limit the amount of candles
     * @param $startTime string request candle information starting from here
     * @param $endTime string request candle information ending here
     * @return array containing the response
     * @throws \Exception
     */
    public function candlesticks(string $symbol, string $interval = "5m", int $limit = null, $startTime = null, $endTime = null)
```

**Comment by PanderBoy18** [comment]
> Added optional limit to chart function, this parameter comes after the callback. Now you can set it to the maximum of 5000
> 
> ```
> $binance_api->chart(["LTCUSDT"], "15m", function($binance_api, $symbol, $chart) {
> // ...
> }, 1000);
> ```
> 
> Available in `v0.5.2` [60b7074](https://github.com/jaggedsoft/php-binance-api/commit/60b7074805e799b4af175a9b1b57ed71dad849a4)

I think you mean a maximum of 1000?!

**Comment by jaggedsoft** [comment]
@PanderBoy18 yes sorry, this is what I meant to say:
### maximum of 1000, default is 500

cheers

![feelspopoman](https://cdn.discordapp.com/emojis/403330647550197773.png "derp")

**Comment by jaggedsoft** [comment]
Yeah just add `000` to the end of your time or multiply by 1000 `* 1000`
the reason is that php time is given in seconds, and javascript time is given in milliseconds. They expect it in javascript format

**Comment by jaggedsoft** [comment]
haven't tested this but I would expect it to work
```php
$limit = 1000;
$startTime = strtotime("2018-01-01) * 1000;
$endTime = strtotime("Sat, 27 Oct 2018 15:42:21 UTC") * 1000;
candlesticks("BTCUSDT", "15m", $limit, $startTime, $endTime)
```

**Comment by PanderBoy18** [comment]
Thanks, that worked for me!
